tracing-attributes/src/lib.rs

```diff
@@ -353,13 +353,31 @@ mod expand;
 /// const fn my_const_function() {}
 /// ```
 ///
+/// When a function returns a [`pin`]ned future and the function and its future have to be inside the
+/// same span, current span can be attached to the returned future like this:
+///
+/// ```
+/// # use tracing_attributes::instrument;
+/// # use std::pin::Pin;
+/// # use std::future::Future;
+/// #[instrument]
+/// fn foo() -> Pin<Box<dyn Future<Output = ()>>> {
+///     tracing::info!("works inside foo");
+///     let f = async {
+///          tracing::info!("works inside future returned by foo");
+///     }.instrument(tracing::Span::current());
```

Comment:
I think it might be nicer to suggest the [`in_current_span()` method](https://docs.rs/tracing/latest/tracing/trait.Instrument.html#method.in_current_span) here --- it's shorthand for this.

```suggestion
///     }.in_current_span();
```